### PR TITLE
Converted `App/Dialogs` to stateless components where applicable

### PR DESCRIPTION
**Pull request by ericadamski**
Part of #518 

Also update some of the `setTimeout(() => ReactDOM.findDOMNode(...), 1)` calls to use React lifecycle methods and ref functions rather than string references.

### CODE REVIEW

**Comment by ericadamski** [comment]
👍

**Comment by skellock** [comment]
Omg, that setTimeout was nasty. Thx for the proper technique.

**Comment by ericadamski** [comment]
@skellock it was pretty nasty 😜

File: packages/reactotron-app/App/Dialogs/StateWatchDialog.js [hunk 1]
```diff
@@ -102,13 +102,16 @@ class StateWatchDialog extends Component {
     session.ui.watchToAdd = e.target.value
   }
 
+  componentDidUpdate() {
```
Comment (M-ZubairAhmed) [inline packages/reactotron-app/App/Dialogs/StateWatchDialog.js]: this is a really good way of doing it @ericadamski

**Comment by M-ZubairAhmed** [comment]
@ericadamski i learned a lot from this PR thank you

**Comment by ericadamski** [comment]
@M-ZubairAhmed I am really glad! Thank you :)

**Pull request merged**